File: DSA_Codesheet/Python/sort012.py
from os import *
from sys import *
from collections import *
from math import *

from sys import stdin,setrecursionlimit
setrecursionlimit(10**7)

def sort012(arr, n) :
    k = 0
    occurences = [0,0]
    for i in range(n):
        if(arr[i] == 0):
            arr[k] = 0
            k+=1
        else:
            occurences[arr[i]-1]+=1
    for i in range(2):
        for j in range(occurences[i]):
            arr[k] = i+1
            k+=1
    pass


#taking inpit using fast I/O
def takeInput() :
	n = int(input().strip())

	if n == 0 :
		return list(), 0

	arr = list(map(int, stdin.readline().strip().split(" ")))

	return arr, n



def printAnswer(arr, n) :
    
    for i in range(n) :
        
        print(arr[i],end=" ")
        
    print()
    
#main

t = int(input().strip())
for i in range(t) :

    arr, n= takeInput()
    sort012(arr, n)
    printAnswer(arr, n)

File: DSA_Codesheet/Cpp/sort012.cpp
#include <bits/stdc++.h> 
void sort012(int *arr, int n)
{
    int k = 0;
    int occurences[2] = {0,0};

    for(int i = 0; i < n; i++) {
        if(arr[i] == 0) {
            arr[k] = 0;
            k++;
        } else {
            occurences[arr[i]-1]++;
        }
    }

    for(int i = 0; i < 2; i++) {
        for(int j = 0; j < occurences[i]; j++) {
            arr[k] = i+1;
            k++;
        }
    }  
}